capstone/src/main/java/com/google/step/clients/MetricsClient.java

```diff
@@ -0,0 +1,288 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.step.clients;
+
+import com.google.appengine.api.datastore.*;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Query.CompositeFilter;
+import java.util.*;
+
+/**
+ * This client is in charge of interacting with metricsServlet and the metrics entities.
+ */
+public class MetricsClient {
+
+    /**
+     * Empty constructor since we don't need any variables, we only use this class for the methods.
+     */
+    public MetricsClient() {}
+
+    private static DatastoreService dataStore = DatastoreServiceFactory.getDatastoreService();
+
+    /**
+     * Gets the current week's page views for one restaurant by specifying a restaurantKey.
+     * @param restaurantKey restaurant key of the restaurant we want to get views.
+     * @param restaurantName restaurant name in case no entity exists for this week's page views.
+     * @return returns a Map<String, Object> of the form:
+     *          {
+     *              "year": <String, the year number>,
+     *              "week": <String, this week's number in relation to the year.>,
+     *              "count": <String, amount of page views the restaurant had.>
+     *          }
+     */
+    public Map<String, Object> getCurrentPageViews(String restaurantName, String restaurantKey) {
+        Map<String, Object> resultMap = new HashMap<>();
+
+        Calendar calendar = Calendar.getInstance();
+        int year = calendar.get(Calendar.YEAR);
+        int week = calendar.get(Calendar.WEEK_OF_YEAR);
+
+        CompositeFilter compositeFilter = new CompositeFilter(Query.CompositeFilterOperator.AND,
+                Arrays.asList(
+                        new FilterPredicate("restaurantKey", Query.FilterOperator.EQUAL, restaurantKey),
+                        new FilterPredicate("year", Query.FilterOperator.EQUAL, year),
+                        new FilterPredicate("week", Query.FilterOperator.EQUAL, week)
+                ));
+
+        Query query = new Query("PageViews").setFilter(compositeFilter);
+        PreparedQuery preparedQuery = dataStore.prepare(query);
+        Entity results = preparedQuery.asSingleEntity();
+
+        // If there are no entities of that restaurant, add one with a count of 0.
+        if (results == null) {
```

Comment:
Is this entire transaction necessary? I'd suggest you just let `putPageView` handle this in all cases. This extra code doesn't really add much value. In fact, this extra code can result in a needless error (if a page view is added at the same time and breaks this transaction).

This would actually be more robust if you just returned the `zero` case if you didn't find anything.

Reply:
The transaction has been taken out and it just returns a WeeklyPageView class with all variables set to 0.